smartsim/experiment.py

```diff
@@ -98,19 +109,14 @@ class Experiment:
     and utilized throughout runtime.
     """
 
-    def __init__(
-        self,
-        name: str,
-        exp_path: t.Optional[str] = None,
-        launcher: str = "local",
-    ):
+    def __init__(self, name: str, exp_path: str | None = None):
```

Comment:
They are 100% equivalent! I just like the latter because:
1) It's slightly less to type, lol
1) I think `T | None` does a better job of communicating that this parameter is nullable than `Optional[T]`

Reply:
Oh! One last caveat!!

The latter syntax is only available with Python >= 3.10 or with the `from __future__ import annotations` flag for the interpreter. But I'm also on a bit of a mission to start moving more of our modules over to using `from __future__ import annotations` anyways!